Chore: Update developer guide to reflect changes to go get (#26646)
grafana/grafana: contribute/developer-guide.md

## Download Grafana

We recommend using the Git command-line interface to download the source code for the Grafana project:

1. Open a terminal and run `git clone https://github.com/grafana/grafana.git`. This command downloads Grafana to a new `grafana` directory in your current directory.
1. Open the `grafana` directory in your favorite code editor.

For alternative ways of cloning the Grafana repository, please refer to [GitHub's cloning a repository](https://docs.github.com/en/github/creating-cloning-and-archiving-repositories/cloning-a-repository) documentation.

> We do not recommend using Go's `go get` to download Grafana. Recent versions of Go have added behavior which isn't compatible with the way the Grafana repository is structured.

Reviewer: ```suggestion
We don't recommend using Go's `go get` to download Grafana. Recent versions of Go have added behavior which isn't compatible with the way the Grafana repository is structured.
```

Author: I'm not a fan of contracting the emphasized part of a sentence. I now took a peek at our [documentation style guide](https://github.com/grafana/grafana/blob/34c2f440dbd701e31faa0949ddfc8921a8c6f460/contribute/style-guides/documentation-style-guide.md) and there isn't anything about this in particular, but we're using both don't and do not within that file.

I'll keep it as I wrote it for now, but I'm open to changing it given a rationale 🙂

Reviewer 2: I also prefer "do not" here, agree with @sakjur.

Reviewer 3: I would also change this to a warning.

> **Warning:** Do not use `go get` to download Grafana. Recent versions...

Author: ```suggestion
**Warning:** Do not use `go get` to download Grafana. Recent versions of Go have added behavior which isn't compatible with the way the Grafana repository is structured.
```

> We do not recommend that you use Go's `go get` to download Grafana. Recent versions of Go have added behavior which isn't compatible with the way the Grafana repository is structured.

Reviewer 2: Nit, but I think the following reads better:
```suggestion
We do not recommend using Go's `go get` to download Grafana. Recent versions of Go have added behavior which isn't compatible with the way the Grafana repository is structured.
```
